qa/tasks/cephfs/test_failover.py

```diff
@@ -706,6 +709,90 @@ def test_rank_stopped(self):
+        # until a threshold. default threshold is too high. it will
+        # unnecssarily consume so much time and resources.
+        self.run_ceph_cmd('config set mds mds_health_cache_threshold 1.000001')
+        # so that there is only active MDS and only 1 health warning is
+        # produced. presence of 2 warning should cause this test to fail
+        self.fs.set_max_mds(1)
+        self.run_ceph_cmd(f'fs set {self.fs.name} allow_standby_replay true')
+        self._confirm_single_replay()
+        self.fs.wait_for_daemons()
+        # The call above (to self.fs.wait_for_daemons()) should ensure we have
+        # only 1 active MDS on cluster
+        active_mds_id = self.fs.get_active_names()[0]
+        sr_mds_id = self.fs.get_standby_replay_names()[0]
+
+        # this should generate MDS cache more than 3M and therefore should
+        # trigger health warning MDS_CACHE_OVERSIZED.
+        open_proc = self.mount_a.open_n_background(".", 4000)
+
+        # actual test begins now...
+        errmsg = ('this test didn\'t generate health warning it was supposed '
+                  'to. health report has been logged above.')
+        with safe_while(sleep=3, tries=10, action=errmsg) as proceed:
+            while proceed():
+                # logging cache generated so far for th sake of easy
+                # debugging in future.
+                self.get_ceph_cmd_stdout(
+                    f'daemon mds.{active_mds_id} cache status')
+
+                health_report = self.get_ceph_cmd_stdout(
+                    'health detail --format json')
+                health_report = json_loads(health_report)
+                try:
+                    health_report['checks']['MDS_CACHE_OVERSIZED']
+                except KeyError:
+                    log.info('warning hasn\'t appeared in health report yet.'
+                             'trying again after some sleep...')
+                    continue
+
+                cache_warn = health_report['checks']['MDS_CACHE_OVERSIZED']\
+                        ['detail']
+                log.info(f'cache_warn - {cache_warn}')
+                # sanity check: "ceph health detail" output should've 2
+                # warnings -- one for active MDS and other for standby-replay
+                # MDS.
+                try:
+                    self.assertEqual(len(cache_warn), 2)
+                except:
+                    log.info(f'{len(cache_warn)} warnings found in log; '
+                              'expected 2. trying again after some sleep...')
+                    continue
+
+                for cw in cache_warn:
+                    msg = cw['message']
+
+                    if f'mds.{sr_mds_id}' not in cw['message']:
+                        continue
+
+                    self.assertNotIn('inodes in use by clients', msg)
+                    self.assertNotIn('stray files', msg)
+
+                    # test was successful, tearing down now...
```

Comment:
It's good practice to close files opened during testing before attempting umount...

Reply:
It probably doesn't matter since we delete entire CephFS and re-spawn MDS for every test but it's still good to do so instead of hitting bugs later.

I checked our QA codebase, we do so the same elsewhere too.

Reply:
> It's good practice to close files opened during testing before attempting umount...

In general, yes. mount::teardown() will kill all background procs.

Reply:
Removed.